ros_cross_compile/ros_cross_compile.py

```diff
@@ -153,17 +187,45 @@ def cross_compile_pipeline(
         default_docker_dir=sysroot_build_context,
         colcon_defaults_file=args.colcon_defaults)
 
+    # stage 1
     if not args.skip_rosdep_collection:
-        gather_rosdeps(
-            docker_client=docker_client,
-            platform=platform,
-            workspace=ros_workspace_dir,
-            skip_rosdep_keys=skip_rosdep_keys,
-            custom_script=custom_rosdep_script,
-            custom_data_dir=custom_data_dir)
-    assert_install_rosdep_script_exists(ros_workspace_dir, platform)
-    create_workspace_sysroot_image(docker_client, platform)
-    run_emulated_docker_build(docker_client, platform, ros_workspace_dir)
+        gather_rosdeps_wrapper((
+            docker_client,
+            platform,
+            ros_workspace_dir,
+            skip_rosdep_keys,
+            custom_rosdep_script,
+            custom_data_dir), dc)
+
+    # get the size of the docker image generated in this stage
+    rosdep_img_size = docker.from_env().images.get('ros_cross_compile:rosdep').attrs['Size']
+    dc.add_datum('gather_rosdeps-img_size', rosdep_img_size / 1000000, 'MB', str(today))
+
+    # stage 2
+    rosdep_script_wrapper((ros_workspace_dir, platform), dc)
+
+    # stage 3
+    create_workspace_sysroot_wrapper((docker_client, platform), dc)
+
+    # get the size of the docker image generated in this stage
+    sysroot_img_size = docker.from_env().images.get(
+        'ubuntu/aarch64-ubuntu-foxy:latest').attrs['Size']
+    dc.add_datum('sysroot_img-size', sysroot_img_size / 1000000, 'MB', str(today))
+
+    # stage 4
+    run_docker_build_wrapper((docker_client, platform, ros_workspace_dir), dc)
+
+    # stop timing
+    total = time.time() - start
+    dc.add_datum('total_pipeline-time', total, 'seconds', str(today))
+
+    # write metrics to file.
+    if args.metric_file_path is not None:
```

Comment:
Remove, if we're going to write the file anyway, we just need to get argparse to default to a logical value. No need to branch out the logic here.